Toolbox/ArcSDM.pyt

```diff
@@ -784,6 +786,186 @@ class CalculateWeights(object):
+            name="evidence_type",
+            datatype="GPString",
+            parameterType="Required",
+            direction="Input",
+        )
+        param_evidence_source_type.filter.list = ["Raster", "None"]
+        param_evidence_source_type.value = "Raster"
+
+        param_evidence_rasters = arcpy.Parameter(
+            displayName="Evidence rasters (same size/extent/projection)",
+            name="evidence_rasters",
+            datatype="GPRasterLayer",
+            parameterType="Optional",
+            direction="Input",
+            multiValue=True,
+        )
+
+        param_evidence_vectors = arcpy.Parameter(
+            displayName="Evidence vectors (CSV/TXT path)",
+            name="evidence_vectors_file",
+            datatype="DEFile",
+            parameterType="Optional",
+            direction="Input",
+        )
+
+        param_output_labelled_pairwise_csi = arcpy.Parameter(
+            displayName="CSV: Pairwise CSI matrix for labelled samples",
+            name="out_labelled_pairwise",
+            datatype="DEFile",
+            parameterType="Required",
+            direction="Output",
+        )
+
+        param_output_centroid_csi_matrix = arcpy.Parameter(
+            displayName="CSV: CSI matrix for class centroids",
+            name="out_centroid_matrix",
+            datatype="DEFile",
+            parameterType="Required",
+            direction="Output",
+        )
+
+        param_evidence_csi_table = arcpy.Parameter(
+            displayName="CSV: CSI table for evidence vectors (when evidence is CSV/TXT)",
+            name="out_evidence_table",
+            datatype="DEFile",
+            parameterType="Optional",
+            direction="Output",
+        )
+
+        param_output_raster_directory = arcpy.Parameter(
+            displayName="Folder to save output CSI rasters",
+            name="out_raster_folder",
+            datatype="DEFolder",
+            parameterType="Optional",
+            direction="Output",
+        )
+
+        return [
+            param_labelled_data_path,
+            param_csv_no_data_sentinel,
+            param_label_field_names,
+            param_coordinate_field_names,
+            param_feature_field_names,
+            param_evidence_source_type,
+            param_evidence_rasters,
+            param_evidence_vectors,
+            param_output_labelled_pairwise_csi,
+            param_output_centroid_csi_matrix,
+            param_evidence_csi_table,
+            param_output_raster_directory
+        ]
+
+    def isLicensed(self):
+        """Set whether tool is licensed to execute."""
+        return arcpy.CheckExtension("Spatial") == "Available"
+
+    def updateParameters(self, parameters):
+        # Toggle visibility based on evidence type
+        ev_type = parameters[5].valueAsText  # evidence source type
+        # Indices:
+        # 6: evidence rasters
+        # 7: evidence vectors
+        # 10: output evidence table

```

Comment:
The comment for 9 (centroid CSI matrix) is a bit misleading - maybe it got out of date at some point? These could be read into variables to improve readability.